src/commands/commandRegistry.ts

```diff
@@ -20,8 +20,10 @@ export class CommandRegistry {
    * @param {@link ICommand} command
    */
   register(phrase: string, command: ICommand) {
-    // TODO: check for existing command with this name. throw error if collison
-    const existingCommand = _.find(Array.from(this._commands.values()), (cmd) => (cmd.name = command.name)) as ICommand;
+    const existingCommand = _.find(
+      Array.from(this._commands.values()),
+      (cmd) => cmd.constructor.name == command.name
```

Comment:
Biggest bug of them all!

I had `=` instead of `==` so it was always `true` therefore was never finding the command I thought it was. It was just grabbing any command.